test/e2e/tests/test_db_proxy.py

```diff
@@ -0,0 +1,100 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the RDS API DBProxy resource
+"""
+
+import time
+
+import pytest
+
+from acktest.k8s import resource as k8s
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_rds_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e import condition
+from e2e import db_proxy
+from e2e.fixtures import k8s_secret
+from e2e import tag
+
+RESOURCE_PLURAL = 'dbproxies'
+
+DELETE_WAIT_AFTER_SECONDS = 120
+
+# Time we wait after resource becoming available in RDS and checking the CR's
+# Status has been updated.
+CHECK_STATUS_WAIT_SECONDS = 60*4
+
+MODIFY_WAIT_AFTER_SECONDS = 20
+
+
+@service_marker
+@pytest.mark.canary
+class TestDBProxy:
+
+    def test_crud_postgresql_proxy(
+            self,
+    ):
+        db_proxy_id = "my-test-proxy"
+        db_proxy_engine_family = "POSTGRESQL"
+        # The IAM role and secrect below has a complext dependency chain and we can hard code it for now
+        # It needs create one rds instance -> create aws secret based on it -> create IAM role based on this secret
+        # I don't have a better way to fit this dependency chain in current rds controller yet
+        iam_role_arn = "arn:aws:iam::274006911594:role/Admin"
```

Comment:
We have a bootstrappable resource to create iam_role_arns

Reply:
will try it in next rev